crates/forge/tests/cli/script.rs

```diff
@@ -2645,3 +2645,32 @@ Script ran successfully.
 
 "#]]);
 });
+
+// Tests that script reverts if it uses `address(this)`.
+forgetest_init!(should_revert_on_address_opcode, |prj, cmd| {
+    prj.add_script(
+        "ScriptWithAddress.s.sol",
+        r#"
+        import {Script, console} from "forge-std/Script.sol";
+
+    contract ScriptWithAddress is Script {
+        function run() public view {
+            console.log("script address", address(this));
+        }
+    }
+    "#,
+    )
+    .unwrap();
+
+    cmd.arg("script").arg("ScriptWithAddress").assert_failure().stdout_eq(str![[r#"
+...
+[..] ERROR forge::script: Usage of `address(this)` detected in script contract. Script contracts are ephemeral and their addresses should not be relied upon.
+...
+
```

Comment:
Errors are expected to be logged in `stderr`

Reply:
fixed in https://github.com/gap-editor/foundry/commit/c01523e61147899f8925f3105b4bcdc1536fb6fa